File: cpp/01-commands/explain/LifeGame/Game.cpp
#include "pch.h"
#include "Game.h"

Game::Game(Grid&& grid)
    : _grid(std::move(grid)), _isFinished(false)
{}

bool Game::IsFinished() const
{
    return _isFinished;
}

void Game::Run()
{
    Grid nextGrid = _grid.NextState();

    if (_grid.IsEqual(nextGrid))
    {
        _isFinished = true;
    }

    _grid = std::move(nextGrid);
}

bool Game::GridIsEmpty() const
{
    return _grid.IsEmpty();
}

File: python/01-commands/explain/LifeGame/game.py
from grid import Grid

class Game:
    def __init__(self, grid: Grid):
        self.grid = grid
        self.is_finished = False

    def run(self):
        next_grid = self.grid.next_state()

        if self.grid.is_equal(next_grid):
            self.is_finished = True

        self.grid = next_grid

    def grid_is_empty(self):
        return self.grid.is_empty
